### PR TITLE
travis: Use rustup to install the i686 cross target.

### PR DESCRIPTION
**Pull request by rillian**
This is easier to understand than using the external wrapper
script, with everything defined in-tree. We lose the hack
around doc-tests not working with cross-compilation, however.

Define TARGET and FEATURES only if they are different
from the defaults. Use rustup to install the alternate
target support. Use posix shell parameter expansion to
prefix the --target and --features cargo switches if the
corresponding variables are set.

Use folded blocks to improve formatting. The '>' prefix
character folds the subsequent indented block into a single
string with newlines removed. This lets us add and remove
switches cleanly without long lines or continuation. Note
however that the whole block must use the same indentation.

Remove some keys which just declared default values.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,32 +24,41 @@ addons: &apt_64
...
   - rust: nightly
-    env: ARCH=i686
+    env:
+      - TARGET=i686-unknown-linux-gnu
     addons: *apt_32
   - rust: nightly
-    env: ARCH=i686 CARGOOPTS="--features pulse-dlopen"
+    env:
+      - TARGET=i686-unknown-linux-gnu
+        - FEATURES=pulse-dlopen
```
Comment (djg) [inline .travis.yml]: Indentation.  I set up all the - items with the - aligned to the parent indentation, not indented by 2 spaces.  For example, see:
```yaml
apt:
  packages:
  - gcc-multilib
```

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,19 @@
 language: rust
-sudo: false
```
Comment (djg) [inline .travis.yml]: Does this need to be removed? I added it to get the "container" infrastructure for Travis-CI.

Comment (rillian) [inline .travis.yml]: I believe `sudo: false` is the default, at least for the precise images. The documenation isn't consistent about what's required, of course. :/

I didn't see any difference in the worker description section or in dispatch speed with or without it.

File: .travis.yml (hunk 1 above)
Comment (rillian) [inline .travis.yml]: Oops, sorry. I'll fix. I want to add some empty lines to separate more of the top-level keys without indentation though.

**Pull request merged**